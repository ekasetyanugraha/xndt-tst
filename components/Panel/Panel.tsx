import React from 'react';

type Color = 'primary' | 'link' | 'info' | 'success' | 'warning' | 'danger';

interface Props {
  color?: Color;
  title?: string;
  children?: JSX.Element | JSX.Element[];
}

export default function Panel({ color = 'primary', title, children }: Props): JSX.Element {
  return (
    <div className={`panel is-${color}`}>
      {title && <p className="panel-heading">{title}</p>}

      <div className="p-4">{children}</div>
    </div>
  );
}
