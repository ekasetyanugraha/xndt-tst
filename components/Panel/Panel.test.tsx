import React from 'react';
import { render, screen } from '@testing-library/react';
import Panel from '.';

describe('Component: Panel', () => {
  it('should show panel with primary color when color is not provided', () => {
    render(<Panel />);

    expect(screen.getByTestId('panel').className).toMatchInlineSnapshot(`"panel is-primary"`);
  });

  it('should show panel with provided color', () => {
    render(<Panel color="danger" />);

    expect(screen.getByTestId('panel').className).toMatchInlineSnapshot(`"panel is-danger"`);
  });

  it('should show panel with provided color', () => {
    render(<Panel color="danger" />);

    expect(screen.getByTestId('panel').className).toMatchInlineSnapshot(`"panel is-danger"`);
  });

  it('should show children properly', () => {
    render(
      <Panel title="Test Panel Title">
        <p>Test Children</p>
      </Panel>,
    );

    expect(screen.getByTestId('panel')).toMatchSnapshot();
  });
});
