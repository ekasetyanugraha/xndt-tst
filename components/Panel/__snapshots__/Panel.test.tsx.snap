// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Panel should show children properly 1`] = `
<div
  class="panel is-primary"
  data-testid="panel"
>
  <p
    class="panel-heading"
  >
    Test Panel Title
  </p>
  <div
    class="p-4"
  >
    <p>
      Test Children
    </p>
  </div>
</div>
`;
